internal/service/s3/bucket_object_test.go

```diff
@@ -0,0 +1,2147 @@
+	var obj1 s3.GetObjectOutput
+	resourceName := "aws_s3_bucket_object.object"
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, s3.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckBucketObjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBucketObjectConfig_defaultBucketSSE(rName, "stuff"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckBucketObjectExists(resourceName, &obj1),
+					testAccCheckBucketObjectBody(&obj1, "stuff"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccS3BucketObject_ignoreTags(t *testing.T) {
+	var obj s3.GetObjectOutput
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_s3_bucket_object.object"
+	key := "test-key"
+	var providers []*schema.Provider
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ErrorCheck:        acctest.ErrorCheck(t, s3.EndpointsID),
+		ProviderFactories: acctest.FactoriesInternal(&providers),
+		CheckDestroy:      testAccCheckBucketObjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {},
+				Config: acctest.ConfigCompose(
+					acctest.ConfigIgnoreTagsKeyPrefixes1("ignorekey"),
+					testAccBucketObjectConfig_withNoTags(rName, key, "stuff")),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckBucketObjectExists(resourceName, &obj),
+					testAccCheckBucketObjectBody(&obj, "stuff"),
+					testAccCheckBucketObjectUpdateTags(resourceName, nil, map[string]string{"ignorekey1": "ignorevalue1"}),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+					testAccCheckBucketObjectCheckTags(resourceName, map[string]string{
+						"ignorekey1": "ignorevalue1",
+					}),
+				),
+			},
+			{
+				PreConfig: func() {},
+				Config: acctest.ConfigCompose(
+					acctest.ConfigIgnoreTagsKeyPrefixes1("ignorekey"),
+					testAccBucketObjectConfig_withTags(rName, key, "stuff")),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckBucketObjectExists(resourceName, &obj),
+					testAccCheckBucketObjectBody(&obj, "stuff"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "3"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key1", "A@AA"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key2", "BBB"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key3", "CCC"),
+					testAccCheckBucketObjectCheckTags(resourceName, map[string]string{
+						"ignorekey1": "ignorevalue1",
+						"Key1":       "A@AA",
+						"Key2":       "BBB",
+						"Key3":       "CCC",
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAccS3BucketObject_bucketObjectBucket(t *testing.T) {
+	var obj1 s3.GetObjectOutput
+	resourceName1 := "aws_s3_bucket_object.test"
+	resourceName2 := "aws_s3_object.test"
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
```

Comment:
- If a practitioner takes config and searches s3_bucket_object, replaces s3_object, and applies, TF will destroy any such objects and recreate them.
- If you have an object you don’t want to be recreated, you have options: 1) leave it as s3_bucket_object for now, 2) import the resource into s3_object